### PR TITLE
feat(backend): gym kiosk CRUD and public kiosk read

### PR DESCRIPTION
**Pull request by marcodejongh**
## What & why

PR E of the gym-kiosk-dashboard epic: the backend GraphQL surface for gym kiosks (the preset smart-TV wall dashboard).

Adds:
- **`gymKiosk(gymSlug, kioskSlug?)`** — public, rate-limited, no-login read. Public gym → anyone; private gym → only a viewer who can edit it (else `null`, indistinguishable from missing). No `kioskSlug` → the gym's oldest live kiosk. The stored `layout` is read leniently (a corrupt/future-version row degrades to empty, never errors); `boards` is the resolved slot list — dead/unlinked slots dropped, and non-public boards filtered out for viewers without access (the private board stays in `layout`, off `boards`, so the client degrades the preset).
- **`gymKiosks(gymUuid)`** — manage-UI list, gym-edit-access gated, oldest first.
- **`createGymKiosk` / `updateGymKiosk` / `deleteGymKiosk`** — all `requireGymEditAccess`. Create derives a unique slug from the name (or takes an explicit one), enforces `MAX_KIOSKS_PER_GYM` (10) and per-gym slug uniqueness. Update strict-validates `layout` against `@boardsesh/kiosk`'s `KioskLayoutSchema`, persists the schema-parsed output (backend is the schema authority — unknown keys stripped at every level), and rejects any slot/leaderboard board that isn't an alive board linked to the kiosk's gym. Delete is a soft delete that frees the slug.

Board resolution reuses the shared `enrichBoards` `boardId` gate (public, or the viewer can edit the board) rather than the `gymBoards` listing rule — so unlisted-but-public boards still render on a kiosk; only non-public ones are filtered for anon.

Also hardens `isUniqueViolation` to walk the `DrizzleQueryError.cause` chain (drizzle 0.45 wraps driver errors), so the per-gym slug-uniqueness catch actually fires; a raw pg error still matches at depth 0.

## Stack

#3627 (`feat/gym-kiosk-db-schema`: `gym_kiosks` table + `@boardsesh/kiosk` v2 schema) → #3631 (`feat/gym-branding-logo-boards`: branding, `gymBoards`, `UserBoard.boardId`) → **this** (`feat/gym-kiosk-crud`).

## Validation

- `vp check` — format clean; the only lint errors are 3 pre-existing `no-floating-promises` in `packages/db/src/queries/__tests__/tick-offset-inference.test.ts` (untouched — zero diff under `packages/db`).
- `vp run typecheck` — passes.
- `packages/backend/src/__tests__/gym-kiosks.test.ts` — 26/26 passing (single-file run).
- Zero diffs under `packages/db`.

## Release Notes

- [x] No release note needed (internal / technical change)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01MaW2zXJkhwXJhyxYzSv9iH

### CODE REVIEW

File: packages/backend/src/graphql/resolvers/social/gym-kiosks.ts [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    const userId = ctx.userId!;
+
+    const gym = await requireGymEditAccess(validated.gymUuid, userId);
+
+    // Enforce the per-gym kiosk cap (live kiosks only; soft-deleted rows free a slot).
+    const [countRow] = await db
+      .select({ value: count() })
+      .from(dbSchema.gymKiosks)
+      .where(and(eq(dbSchema.gymKiosks.gymId, gym.id), isNull(dbSchema.gymKiosks.deletedAt)));
+    if (Number(countRow?.value ?? 0) >= MAX_KIOSKS_PER_GYM) {
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/backend/src/graphql/resolvers/social/gym-kiosks.ts]: Fixed in 3d2f811b5 — the cap check + insert now run in a transaction behind a `SELECT ... FOR UPDATE` on the gym row, so same-gym creates serialize and the loser re-counts to a clean BAD_USER_INPUT. A deterministic two-racers test asserts exactly one winner and a live count of exactly MAX_KIOSKS_PER_GYM.

File: packages/shared/graphql/src/operations/gym-kiosks.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+// Query/Mutation Variable Types
+// ============================================
+
+export type GetGymKioskQueryVariables = {
+  gymSlug: string;
+  kioskSlug?: string | null;
+};
+
+export type GetGymKioskQueryResponse = {
+  gymKiosk: GymKiosk | null;
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/shared/graphql/src/operations/gym-kiosks.ts]: Fixed in 3d2f811b5 — the response types now use `GymKioskOperationGym` (a Pick of exactly the GYM_KIOSK_GYM_FIELDS selection) via `GymKioskOperationResult`, so reading an unselected Gym field is a compile error instead of undefined at runtime.

**Comment by marcodejongh** [comment]
Re the Claude review rounds (initial + [UPDATED]):

1. **Kiosk cap race (TOCTOU)** — fixed in 3d2f811b5. The count check + insert now run in a transaction behind a `SELECT ... FOR UPDATE` lock on the gym row; concurrent same-gym creates serialize, the loser re-counts and gets BAD_USER_INPUT. Covered by a deterministic concurrency test (two racers at cap−1 → exactly one wins, live count = cap).
2. **N+1 enrichGym in gymKiosks** — fixed in 3d2f811b5. `resolveKioskView` now receives the enriched gym payload; each resolver enriches the gym exactly once, so the manage list does one enrichGym for all its kiosks.
3. **`Record<string, unknown>` updateValues** — fixed in 3d2f811b5. Typed as `Partial<typeof dbSchema.gymKiosks.$inferInsert>`, so a key typo is a compile error rather than a silently ignored column.
4. **Missing updateGymKiosk slug-collision test** — was already added in 686c88a1e ("surfaces an update-path duplicate slug as a clean error"), exercising the update-side KIOSK_SLUG_ALREADY_EXISTS catch end-to-end.
